stdlib/2and3/contextlib.pyi

```diff
@@ -91,3 +91,6 @@ if sys.version_info >= (3, 7):
         def pop_all(self: _S) -> _S: ...
         def aclose(self) -> Awaitable[None]: ...
         def __aenter__(self: _S) -> Awaitable[_S]: ...
+
+if sys.version_info >= (3, 7):
+    def nullcontext(enter_result: Optional[_T] = ...) -> ContextManager[_T]: ...
```

Comment:
Using `Optional` in the argument "strips" `None` from the argument type, although the argument type is returned as is. Also, when no argument is supplied, `_T` can't be determined by the type checker. I think we need an overload for that case.

Reply:
Test case:

```python
from contextlib import nullcontext
from typing import Optional

x: Optional[int]
with nullcontext(x) as t:
    reveal_type(t)  # should print Union[int, None]
with nullcontext() as t:
    reveal_type(t)  # should print None
```